go/cmd/dolt/commands/sqlserver/server.go

```diff
@@ -135,7 +135,7 @@ func Serve(
 		DoltCfgDirPath: serverConfig.CfgDir(),
 		ServerUser:     serverConfig.User(),
 		ServerPass:     serverConfig.Password(),
-		ServerHost:     serverConfig.Host(),
+		ServerHost:     "%",
```

Comment:
This isn't right. We don't want to give up on the -H option entirely for network routing, which this does. We just want to ignore the host name when creating the first user.